weblab/entities/tests/test_views.py

```diff
@@ -98,22 +113,85 @@ def test_create_protocol_requires_permissions(self, user, client):
+        model = recipes.model.make()
+        add_version(model)
+        commit = next(model.commits)
+        response = client.get('/entities/models/{}/versions/{}'.format(model.pk, commit.hexsha))
+        assert response.status_code == 200
+        assert response.context['version'] == commit
+
+        response = client.get('/entities/models/{}/versions/latest'.format(model.pk))
+        assert response.status_code == 200
+        assert response.context['version'] == commit
+
+
+@pytest.mark.django_db
+class TestEntityVersionList:
+    def test_view_entity_version_list(self, client, user):
+        model = recipes.model.make()
+        add_version(model)
+
+        response = client.get('/entities/models/{}/versions/'.format(model.pk))
+        assert response.status_code == 200
+        assert response.context['versions'] == [(None, next(model.commits))]
+
+
+@pytest.mark.django_db
+class TestEntityList:
+    def test_lists_my_models(self, client, user):
+        models = recipes.model.make(_quantity=2, author=user)
+        response = client.get('/entities/models/')
+        assert response.status_code == 200
+        assert list(response.context['object_list']) == models
+
+    def test_lists_my_protocols(self, client, user):
+        protocols = recipes.protocol.make(_quantity=2, author=user)
+        response = client.get('/entities/protocols/')
+        assert response.status_code == 200
+        assert list(response.context['object_list']) == protocols
+
+
 @pytest.mark.django_db
 class TestVersionCreation:
+    def test_new_version_form_includes_latest_version(self, client, user):
+        model = recipes.model.make()
+        add_version(model)
+        add_permission(user, 'create_model_version')
+        commit = next(model.commits)
+        response = client.get('/entities/models/%d/versions/new' % model.pk)
+        assert response.status_code == 200
+        assert response.context['latest_version'] == commit
+
+    def test_no_latest_version(self, client, user):
+        add_permission(user, 'create_model_version')
+        model = recipes.model.make()
+        response = client.get('/entities/models/%d/versions/new' % model.pk)
+        assert response.status_code == 200
+        assert 'latest_version' not in response.context
+
     def test_create_model_version(self, user, client):
         add_permission(user, 'create_model_version')
-        model = ModelEntity.objects.create(
-            name='mymodel',
-            visibility='public',
-            author=user,
-        )
-        EntityFile.objects.create(
-            entity=model,
+        model = recipes.model_file.make(
+            entity__author=user,
             upload=SimpleUploadedFile('model.txt', b'my test model'),
             original_name='model.txt',
-        )
+        ).entity
         response = client.post(
-            '/entities/models/' + str(model.pk) + '/versions/new',
+            '/entities/models/%d/versions/new' % model.pk,
```

Comment:
I thought the `%` operator was being phased out? https://www.python.org/dev/peps/pep-3101/

Reply:
There is a new `.format` approach, but `%` is not yet deprecated as far as I know.

Reply:
I think % often is the most concise option. (until you get to Python 3.6 and start using f-strings :)